marshal.go

```diff
@@ -144,3 +145,157 @@ func (i InstanceMetadata) MarshalXML(e *xml.Encoder, start xml.StartElement) err
 	// flush to ensure tokens are written
 	return e.Flush()
 }
+
+// startLocalName creates a start-tag of an XML element with the given local name and no namespace name.
+func startLocalName(local string) xml.StartElement {
+	return xml.StartElement{Name: xml.Name{Space: "", Local: local}}
+}
+
+type metadataMap map[string]string
+
+// MarshalXML is a custom XML marshaler for metadataMap, mapping each metadata name/value pair to a
+// correspondingly named XML element with the pair's value as character data content.
+func (m metadataMap) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if err := e.EncodeToken(start); err != nil {
+		return err
+	}
+
+	for k, v := range m {
+		if err := e.EncodeElement(v, startLocalName(k)); err != nil {
+			return err
+		}
+	}
+
+	return e.EncodeToken(start.End())
```

Comment:
The documentation mentions _creating an `Encoder`_:
> Callers that create an `Encoder` and then invoke `EncodeToken` directly, without using `Encode` or `EncodeElement`, need to call `Flush` when finished to ensure that the XML is written to the underlying writer.

Here, I didn't create the `Encoder`; it's created implicitly over in `fargo.marshal` in file _net.go_. Note that [`xml.Marshal` creates an `Encoder` and calls `Encode` on it](https://golang.org/src/encoding/xml/marshal.go?s=3004:3047#L62), which in turn [flushes when it's done writing](https://golang.org/src/encoding/xml/marshal.go?s=5858:5905#L144). Hence I don't think that calling `Flush` explicitly here is necessary.